### DISCUSSION TITLE
Highlight / Annotation system is very limited- suggestions for slight extension of the current system

### DISCUSSION THREAD

**Discussion by MagHz** [sumatrapdfreader/sumatrapdf#2809] [Ideas]
Hello,
I am aware that SumatraPDF's goal is to be a pdf _reader_ rather than a pdf editor. Yet - add annotation/highlight functionality exists, so obviously there's at least some interest here.
Unfortunately, current functionality is extremely limited, to the point I found it insufficient and went looking for alternatives. I think extending the current system into one that's just slightly more robust would improve it significantly.

Features I'd like to see: 
- change highlight color/alpha of a given annotation without digging through .txt file
- being able to save a few color/alpha presets for a given annotation
- being able to draw a colored box at(incl. color/alpha settings)

Essentially I installed Foxit Reader just for highlighting and it turns out you can accomplish quite a few things with just those basics - and "click button to change rgba in settings" and "draw a colored rectangle" seem like it wouldn't be too much work, looking at SumatraPDF's current functionality.
Pictured: an example of what I'd like to accomplish with SumatraPDF
![sumatraPDF-annotation-example](https://user-images.githubusercontent.com/12188949/179790089-a8ecf3be-fee9-4e21-96be-d8c3cac2db1d.png)

Thank you for your work!

**Comment by GitHubRulesOK** [comment]
all those and much more can be done by MuPDF engine try [MuPDF-GL](https://www.mupdf.com/releases/) to adjust those values, its just a question of who can devote their free time without failing?
